### PR TITLE
DCOS-19315: remove force updates

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
## Testing

- Please click through all pages that might be affected (I don't want to give detailed pages, because if I miss one you miss them too) and see if they still operate like normal. 

## Trade-offs

- Decided to only change this on a level in the component tree that effects a lot of element on rerenders, this should maximize the impact of this PR
- I could not trace back every removed `forceUpdate`, but I still think we should ship them. If they hide an obscure bug, then we need to create better integration tests to catch that in the future. I see no other way, except keeping them and limiting the performance for all users in the future.
- I choose to use the component state in a very trivial way, as this is IMHO also about removing magic

## Screenshots

|  Page | Before  | After  |
|---|---|---|
| Dashboard  |  Bildschirmfoto 2019-03-19 um 16 39 45 |  Bildschirmfoto 2019-03-19 um 16 38 39 |
| Deployments | Bildschirmfoto 2019-03-19 um 16 41 03  | Bildschirmfoto 2019-03-19 um 16 42 18  |
| Nodes Agents  | Bildschirmfoto 2019-03-19 um 16 43 38  |  Bildschirmfoto 2019-03-19 um 16 44 38 |
| Nodes Detail| Bildschirmfoto 2019-03-19 um 16 45 45 | Bildschirmfoto 2019-03-19 um 16 46 43 |
| Service Detail | Bildschirmfoto 2019-03-19 um 16 48 13 | Bildschirmfoto 2019-03-19 um 16 51 40 |

### CODE REVIEW

File: plugins/nodes/src/js/pages/nodes/NodeDetailPage.js [hunk 1]
```diff
@@ -23,12 +23,12 @@ class NodeDetailPage extends mixin(TabsMixin, StoreMixin) {
     super(...arguments);
 
     this.store_listeners = [
-      { name: "summary", events: ["success"], suppressUpdate: false },
-      { name: "state", events: ["success"], suppressUpdate: false },
+      { name: "summary", events: ["success"], suppressUpdate: true },
+      { name: "state", events: ["success"], suppressUpdate: true },
```
Comment (nLight) [inline plugins/nodes/src/js/pages/nodes/NodeDetailPage.js]: removing those we should make sure we have `on<StoreName>Store<EventName>` functions. I.e. in the component we do have `onStateStoreSuccess` but nothing for summary and nodes

Comment (DanielMSchmidt) [reply plugins/nodes/src/js/pages/nodes/NodeDetailPage.js]: I mean, nothing breaks like this, so maybe we don't need that anymore

Comment (nLight) [reply plugins/nodes/src/js/pages/nodes/NodeDetailPage.js]: can we please ensure checking the components don't rely on data from the stores please?

File: plugins/nodes/src/js/pages/nodes/nodes-table/NodesTableContainer.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ class NodesTableContainer extends mixin(StoreMixin, QueryParamsMixin) {
         name: "nodeHealth",
         suppressUpdate: true
       },
-      { name: "state", events: ["success"], suppressUpdate: false }
+      { name: "state", events: ["success"], suppressUpdate: true }
```
Comment (DanielMSchmidt) [inline plugins/nodes/src/js/pages/nodes/nodes-table/NodesTableContainer.js]: There was a `onStateStoreSuccess` available already

File: plugins/services/src/js/containers/service-detail/ServiceDetail.js [hunk 3]
```diff
@@ -163,7 +163,6 @@ class ServiceDetail extends mixin(TabsMixin) {
       this.hasVolumes()
     ) {
       this.tabs_tabs["/services/overview/:id/volumes"] = "Volumes";
-      this.forceUpdate();
```
Comment (DanielMSchmidt) [inline plugins/services/src/js/containers/service-detail/ServiceDetail.js]: Could not find a root cause for this call

File: plugins/services/src/js/pages/ServicesPage.js [hunk 4]
```diff
@@ -37,7 +37,7 @@ var ServicesPage = createReactClass({
 
   componentWillMount() {
     this.store_listeners = [
-      { name: "notification", events: ["change"], suppressUpdate: false }
```
Comment (DanielMSchmidt) [inline plugins/services/src/js/pages/ServicesPage.js]: No idea which errors [this commit](https://github.com/dcos/dcos-ui/commit/b73c4540f9133b47af0b3ecacab73b3180415452) talks about, I found no data requirement

File: src/js/components/charts/Chart.js [hunk 5]
```diff
@@ -28,7 +28,7 @@ var Chart = createReactClass({
       {
         name: "sidebar",
         events: ["widthChange"],
-        suppressUpdate: false
```
Comment (DanielMSchmidt) [inline src/js/components/charts/Chart.js]: This was already handled

**Comment by nLight** [comment]
The mixin by default does force updates and in some places we don't have the `suppressUpdate` specified thus having `forceUpdate` i.e. https://github.com/dcos/dcos-ui/blob/danielmschmidt/trying-if-tests-break-on-supress-update/src/js/components/Page.js#L84 Do we want fixing those places as well?

**Comment by DanielMSchmidt** [comment]
Snap, my search did not find those. Will go through them 👍

**Comment by DanielMSchmidt** [comment]
I'll skip small and unprioritized pages though :)

**Comment by DanielMSchmidt** [comment]
@nLight Please take another look, I think I got all big pages and impactful spots.

**Comment by nLight** [comment]
code looks solid haven't tested yet

**Comment by DanielMSchmidt** [comment]
Thank you @mperrotti I got this error covered now 👍

**Comment by DanielMSchmidt** [comment]
@mperrotti @nLight Just rebased, could you take another look 👍

**Comment by mperrotti** [comment]
@DanielMSchmidt - I'm still seeing that error :(
However, I just noticed it's also happening on the soak cluster, so it's not related to these changes. Still something we should address though.

Video of error on soak113s: https://cl.ly/6bc0d5239644

**Comment by DanielMSchmidt** [comment]
@nLight @mperrotti Could you reapprove after the rebase?

File: src/js/components/Page.js [hunk 6]
```diff
@@ -58,7 +57,13 @@ PageHeader.propTypes = {
 const Page = createReactClass({
   displayName: "Page",
 
-  mixins: [InternalStorageMixin, StoreMixin],
+  mixins: [StoreMixin],
```
Comment (nLight) [inline src/js/components/Page.js]: I hate `InternalStorageMixin` even more than `StoreMixin`

**Comment by DanielMSchmidt** [comment]
@brandonc @natmegs Hey folks, I invested some time today to debug the root cause and I am unable to even reproduce the issue. Could you maybe check if this is not a preexisting bug and / or take it from here and debug the issue? Neither problem would appear once in like 30 minutes clicking forth and back 🙈

**Comment by DanielMSchmidt** [comment]
Awesome, could you also check @natmegs or may I dismiss your review?

**Pull request merged**